### PR TITLE
Reorganise documentation

### PR DESCRIPTION
**Pull request by geoknee**
See commit messages for rationale.

### CODE REVIEW

**Review by NiloCK — COMMENTED** [review]
With a separate architecture document, I think it's not too noisy to include both the previous components-with-links diagram and the new api-call-lifecycle diagram. They're illustrative in different ways.

File: architecture.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+## go-nitro architecture
+
+The flow of data through go-nitro (running as an independent process) is shown in this diagram:
+
+![go-nitro architecture](./go-nitro%20architecture.png)
+
+0. An API can originate as a remote procedure call (RPC) over https/ws or nats. The RPC call is handled by a server which consumes a go-nitro node as a library.
```
Comment (NiloCK) [inline architecture.md:7]:
```suggestion
0. An API call can originate as a remote procedure call (RPC) over https/ws or nats. The RPC call is handled by a server which consumes a go-nitro node as a library.
```

File: architecture.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+1. The go-nitro `engine` runs in its own goroutine and has a select statement listening for a message from one of:
```
Comment (NiloCK) [inline architecture.md:8]:
```suggestion
1. The go-nitro `engine` runs in its own goroutine and listens for messages from one of:
```

File: architecture.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+   - the consuming application (the go-nitro `node` translates an API call into a send on a go channel, and returns a go channel where a response will later be received)
+   - the `message` service
+   - the `chain` service.
```
Comment (NiloCK) [inline architecture.md:11]:
```suggestion
   - the consuming application - the end-user wishes to perform some action
   - the `message` service - a network peer has requested some cooperative action
   - the `chain` service - a channel we are involved with has had a state change on chain 
```

prior parenthetical probably too detailed and non-illuminating.

File: contributing.md [hunk 4]
```diff
@@ -175,3 +175,28 @@ You may then view `output.log` in VSCode. A typical log line is then:
 ```log
 [14:37:36.517] DEBUG <engine.go:114>: 0x111A00  <  {"engine":"0x111A00","message":"Constructed Engine"}
 ```
+
+### Start RPC servers with Docker
+
+To spin up a docker image with 3 rpc servers and channels pre-populated, run the following:
+
+1. `make docker/build`
+1. `make docker/start`
```
Comment (NiloCK) [inline contributing.md:184]:
```suggestion
2. `make docker/start`
```

**Comment by geoknee** [comment]
[1afa79d](https://github.com/statechannels/go-nitro/pull/1414/commits/1afa79db682284ad4b796da81f5bf9ffbb8c1408)

File: contributing.md [hunk 5]
```diff
@@ -175,3 +175,28 @@ You may then view `output.log` in VSCode. A typical log line is then:
 ```log
 [14:37:36.517] DEBUG <engine.go:114>: 0x111A00  <  {"engine":"0x111A00","message":"Constructed Engine"}
 ```
+
+### Start RPC servers with Docker
+
+To spin up a docker image with 3 rpc servers and channels pre-populated, run the following:
+
+1. `make docker/build`
```
Comment (NiloCK) [inline contributing.md:183]: this instruction seems fine, but also a naked `make` (typo) seemed to do the trick for me as well.

File: contributing.md [hunk 6]
```diff
@@ -175,3 +175,28 @@ You may then view `output.log` in VSCode. A typical log line is then:
...
+1. `make docker/start`
+
+Three rpc go-nitro servers will be available on ports 4005, 4006, and 4007 for Alice, Irene, and Bob. A ledger channel is created between Alice and Irene, and another ledger channel is created between Irene and Bob. A virtual channel is created between Alice and Bob.
+
```
Comment (NiloCK) [inline contributing.md:187]: It's not obvious at this point why you might do this. Maybe adding some sample CLI calls to each client?

... I don't know how to do this!

File: readme.md [hunk 7]
```diff
@@ -5,118 +5,21 @@ go-nitro
...
-3. Start an RPC server for Alice (`0xAAA6628Ec44A8a742987EF3A114dDFE2D4F7aDCE`) listening for RPCs on port `4005`
-4. Start an RPC server for Irene (`0x111A00868581f73AB42FEEF67D235Ca09ca1E8db`) listening for RPCs on port `4006`
-5. Start an RPC server for Bob (`0xBBB676f9cFF8D242e9eaC39D063848807d3D1D94`) listening for RPCs on port `4007`
+## Usage
 
-Stopping the test script will shutdown all RPC servers and `anvil`.
+> ⚠️ Go-nitro is pre-production software ⚠️
 
-To run the script from the `go-nitro` directory run `go run ./scripts/start-rpc-servers.go`
+Go-nitro it can be consumed either as [library code](./node/readme.md) or started in an [independent process](./doc.go) and interfaced with remote procedure calls (recommended).
```
Comment (NiloCK) [inline readme.md:22]:
```suggestion
Go-nitro can be consumed either as [library code](./node/readme.md) or started in an [independent process](./doc.go) and interfaced with remote procedure calls (recommended).
```

**Review by NiloCK — APPROVED** [review]
LGTM! All "opinion" suggestions optional.

File: contributing.md (hunk 6 above)
Comment (geoknee) [inline contributing.md:187]:
```suggestion
Each server may be communicated with over JSON-RPC at a localhost endpoint with an appropriate port. See [these instructions](./doc.go) on how to do that.
```

**Pull request merged**